toolchain/check/scope_stack.h

```diff
@@ -219,6 +225,32 @@ class ScopeStack {
     Set<SemIR::NameId> names = {};
   };
 
+  // A scope in which `return` can be used.
+  struct ReturnScope {
+    // The `FunctionDecl`.
+    SemIR::InstId decl_id;
+
+    // The value corresponding to the current `returned var`, if any. Will be
+    // set and unset as `returned var`s are declared and go out of scope.
+    SemIR::InstId returned_var = SemIR::InstId::None;
+
+    // When processing a scope that's nested inside a `return` scope where
+    // `return` *cannot* be used (for example, a `class` inside a `fn`), this is
+    // the index of the terminating scope. It's set on `Push`, and cleared on
+    // the appropriate `Pop`.
```

Comment:
I had a little trouble following this. What do you think of this instead?

```suggestion
    // When processing a scope where `return` *cannot* be used, that's nested
    // inside this return scope (for example, a `class` inside a `fn`), this is
    // the index of the nested scope. It's set when that scope is pushed, and
    // cleared when it is popped. 
```

Reply:
I think the suggestion loses the distinction that this is only called for the outermost nested scope. Can you take another look at the rephrasing I've done?